src/components/Button/Button.jsx

```diff
@@ -1,17 +1,34 @@
+// @flow
+/* eslint react/prop-types: 0 */
 import cx from 'classnames';
-import React from 'react';
-import PropTypes from 'prop-types';
-import {
-  getDataAttrs,
-  getDataProps }
-from '../../dataUtils';
+import React, { type Node } from 'react';
+import { getDataAttrs } from '../../dataUtils';
 import Loader from '../Loader/Loader.jsx';
 import style from './style.scss';
+import type { Data } from '../../types';
 
 const type = { PRIMARY: 'primary', SECONDARY: 'secondary', TERTIARY: 'tertiary' };
 const size = { SMALL: 'small' };
 
-class Button extends React.Component {
+type Props = {|
+    label: string,
+    onClick: (evt: MouseEvent) => void,
+    type?: string,
+    disabled: boolean,
+    loading: boolean,
+    children: Node,
+    style: Object,
+    isSubmit: boolean,
+    size: string,
+    data: Data,
+|};
+
+class Button extends React.Component<Props> {
+  static defaultProps = {
+    type: type.PRIMARY,
```

Comment:
this actually works?!

Reply:
It does. But Changing it to buttonType for clarity.